clusterloader2/pkg/prometheus/experimental.go

```diff
@@ -75,21 +89,54 @@ func (pc *PrometheusController) trySnapshotPrometheusDisk() (bool, error) {
 		klog.Info(string(s))
 		return true, nil
 	}
-	snapshotName := pdName
+	pc.diskMetadata.name = pdName
+	pc.diskMetadata.zone = zone
+	return true, nil
+}
+
+func (pc *PrometheusController) snapshotPrometheusDiskIfEnabled() error {
+	if enabled, err := pc.isEnabled(); !enabled {
+		return err
+	}
+	// Update cache of Prometheus disk metadata
+	err := wait.Poll(
```

Comment:
NIT: You could just call cachePrometheusDiskMetadataIfEnabled here

Reply:
I prefer to leave it as it is. It seems very similar, but error handling is different.